src/lib/transaction.rs

```diff
@@ -27,6 +21,36 @@ pub struct Transaction {
     pub(crate) payload: TransactionPayload,
 }
 
+impl Transaction {
+    pub fn new(version: u8, tx_type: u8, payload: TransactionPayload) -> Self {
+        Self {
+            version,
+            tx_type,
+            payload,
+        }
+    }
+
+    pub fn into_bytes(&self) -> Vec<u8> {
```

Comment:
If we want to be consistent with the naming convention of Rust,`into_bytes` shouldn't take a reference to `self`, see the full list here: https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

So either we pass `self` instead of `&self` or we change the name of the method.

Reply:
Fixed, passing `&self` and naming `to_bytes`